src/trompequihash/blake/blake2-config.h

```diff
@@ -0,0 +1,72 @@
+/*
```

Comment:
The libsodium blake2b_update calls blake2b_compress only after it has at least 256 input bytes, preventing the extraction of a midstate.
My own copy of blake is modified to call blake2b_compress eagerly and reduces the buffer to 128 bytes.